### PR TITLE
Add findFontFamily util

### PR DESCRIPTION
**Pull request by evilebottnawi**
@jeddy3 @davidtheclark ping

### CODE REVIEW

File: src/utils/findFontFamily.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import postcss from "postcss"
+import postcssValueParser from "postcss-value-parser"
+import {
+  isStandardValue,
+  isVariable,
+} from "./"
+
+// https://www.w3.org/TR/css-fonts-3/#font-variant-prop
+const fondShorthandKeywords = new Set([
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: font with a "t" :)

File: src/utils/findFontFamily.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      index++
+      realIndex++
+
+      // Ignore non standard syntax
+      if (!isStandardValue(value)) { continue }
+      // Ignore variables
+      if (isVariable(value)) { continue }
+      // Ignore numbers with units
+      if (postcssValueParser.unit(value)) { continue }
+      // Ignore keywords for other animation parts
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: "font parts" right?

File: src/utils/findFontFamily.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      if (fondShorthandKeywords.has(value)) { continue }
+      // Ignore functions
+      if (value.indexOf("(") !== -1) { continue }
+      // If we next string
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: Could you add a more clear comment here?

File: src/utils/findFontFamily.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  "bold", "lighter", "bolder",
+  // font-stretch
+  "semi-condensed", "condensed", "extra-condensed", "ultra-condensed",
+  "semi-expanded", "expanded", "extra-expanded", "ultra-expanded",
+  // font-size
+  "xx-small", "x-small", "small", "medium", "large", "x-large", "xx-large", "larger", "smaller",
+])
+
+/**
+ * Get the font-families and indexes of position font-family
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: How about 

```
Get the font-families within a `font` shorthand property value.

Also returns the index of that each font-family found within the `font` value's list.
```

File: src/utils/__tests__/findFontFamily-test.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import findFontFamily from "../findFontFamily"
+
+test("findFontFamily", t => {
+  t.deepEqual(findFontFamily("12pt/10pt sans-serif"), [{ name: "sans-serif", index: 1 }])
+  t.deepEqual(findFontFamily("bold italic 110% serif"), [{ name: "serif", index: 3 }])
+  t.deepEqual(findFontFamily("normal small-caps 12px/14px fantasy"), [{ name: "fantasy", index: 3 }])
+  t.deepEqual(
+    findFontFamily("italic bold 12px/30px Georgia, serif"),
+    [
+      { name: "Georgia", index: 3 },
```
Comment (davidtheclark) [inline src/utils/__tests__/findFontFamily-test.js]: Why is it returning the index within the list instead of the index in the string?

Comment (evilebottnawi) [inline src/utils/__tests__/findFontFamily-test.js]: @davidtheclark right, rewrite it on index of string

Comment (evilebottnawi) [inline src/utils/__tests__/findFontFamily-test.js]: @davidtheclark `index` should starting with `0` value or `1`?

Comment (davidtheclark) [inline src/utils/__tests__/findFontFamily-test.js]: 0

**Comment by evilebottnawi** [comment]
It was not easy :smile: 
/cc @davidtheclark @jeddy3

File: src/rules/font-family-name-quotes/index.js [hunk 6]
```diff
@@ -54,15 +52,22 @@ export default function (expectation) {
     if (!validOptions) { return }
 
     root.walkDecls(/^font-family$/i, decl => {
```
Comment (davidtheclark) [inline src/rules/font-family-name-quotes/index.js]: Since this uses your nifty new `findFontFamily`, can we also make it check `font` declarations, or were you saving that for later?

File: src/utils/findFontFamily.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  basicKeywords,
+  fontStyleKeywords,
+  fontVariantKeywords,
+  fontWeightKeywords,
+  fontStretchKeywords,
+  fontSizeKeywords,
+  lineHeightKeywords,
+} from "../reference/keywordSets"
+
+function mergeNodeByValue(firstNode, secondNode, charactersBetween) {
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: Let's rename `joinValueNodes`.

File: src/rules/font-family-name-quotes/index.js (hunk 6 above)
Comment (evilebottnawi) [inline src/rules/font-family-name-quotes/index.js]: @davidtheclark i think later. Now we have issue and be cool fix it.

File: src/utils/findFontFamily.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+  let needMergeNodesByValue = false
+  let mergeCharacters = null
+
+  valueNodes.walk((valueNode) => {
+    if (valueNode.type === "function") { return false }
+    if (valueNode.type !== "word"
+      && valueNode.type !== "string"
+      && valueNode.type !== "space"
+      && valueNode.type !== "div"
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: I think it's a little more clear to make a Set of `nodeTypesToCheck` and check if the node's type is included.

File: src/utils/findFontFamily.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    if (fontStyleKeywords.has(value)
+      || fontVariantKeywords.has(value)
+      || fontWeightKeywords.has(value)
+      || fontStretchKeywords.has(value)
+      || fontSizeKeywords.has(value)
+      || lineHeightKeywords.has(value)
+    ) { return }
+    // Ignore numbers with units
+    const unit = getUnitFromValueNode(valueNode)
+    if (unit && /[a-z%]/i.test(unit)) { return }
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: What is this line protecting against?

File: src/utils/findFontFamily.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    if ((valueNode.type === "space" || (valueNode.type === "div" && valueNode.value !== ","))
+      && fontFamilies[fontFamilies.length - 1]
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: Could you just check if the array is empty?

File: src/utils/findFontFamily.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+      || fontVariantKeywords.has(value)
+      || fontWeightKeywords.has(value)
+      || fontStretchKeywords.has(value)
+      || fontSizeKeywords.has(value)
+      || lineHeightKeywords.has(value)
+    ) { return }
+    // Ignore numbers with units
+    const unit = getUnitFromValueNode(valueNode)
+    if (unit && /[a-z%]/i.test(unit)) { return }
+    if ((valueNode.type === "space" || (valueNode.type === "div" && valueNode.value !== ","))
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: A comment would be helpful here: `Detect when a space or comma is dividing a list of font-families, and save the joining character."

File: src/utils/findFontFamily.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+    } else if (valueNode.type === "space" || valueNode.type === "div") {
+      return
+    }
+
+    const fontFamily = valueNode
+
+    if (needMergeNodesByValue) {
+      mergeNodeByValue(fontFamilies[fontFamilies.length - 1], valueNode, mergeCharacters)
+      needMergeNodesByValue = false
+      mergeCharacters = null
```
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: Why do these need to be reset?

**Comment by davidtheclark** [comment]
Nice work! I think the tests show that it's doing the job, and that's great. My comments were just about clarity.

File: src/utils/findFontFamily.js (hunk 9 above)
Comment (evilebottnawi) [inline src/utils/findFontFamily.js]: @davidtheclark font like unit `Hawaii 5px`

File: src/utils/findFontFamily.js (hunk 12 above)
Comment (evilebottnawi) [inline src/utils/findFontFamily.js]: @davidtheclark For comma separate values - `Lucida Grande, Arial, sans-serif`

File: src/utils/findFontFamily.js (hunk 9 above)
Comment (davidtheclark) [inline src/utils/findFontFamily.js]: What issue is this related to? Can you provide a link?

Comment (davidtheclark) [inline src/utils/findFontFamily.js]: What specifically is `/[a-z%]/i.test(unit)` about?

**Comment by davidtheclark** [comment]
This PR needs to be rebased, and needs to avoid the spread operator. Then we can merge.
